### PR TITLE
[NETBEANS-1151] Fix unit test files of php.editor

### PR DESCRIPTION
**Pull request by junichi11**
https://issues.apache.org/jira/browse/NETBEANS-1151
Enable unit tests of php.editor again: see #720

- nette.min.php: New BSD License https://github.com/nette/nette/blob/master/license.md
- testEmptyStatementHint.php: New BSD License https://github.com/nette/nette/blob/master/license.md
- performance.php: Simple Machines License (SMF 1.0 and 1.1) https://www.simplemachines.org/about/smf/license.php
- TextSearchQuery.php: GPL v2 https://sourceforge.net/p/phpwiki/code/HEAD/tree/trunk/LICENSE
- html04.php: Recreate it

html04.php is related to this issue: https://netbeans.org/bugzilla/show_bug.cgi?id=175229, we should also fix html01, 02. 03.php.

I'm not sure whether we can include performance.php.
TextSearchQuery.php is GPL, so download as external binaries?

### CODE REVIEW

**Comment by junichi11** [comment]
I'll try to add TextSearchQuery.php as external binaries later :)

**Comment by junichi11** [comment]
I've added TextSearchQuery-1.28.zip to the binaries-list. Its license is GPL. So we get the following message:
```
php.editor/external/TextSearchQuery-1.28-license.txt has a GPL-family license but is either not covered by the Classpath Exception, or is not compile-time/optional only.
```
@matthiasblaesing Could you improve the VerifyLibsAndLicenses if it's OK to use a file licensed under GPL as test data?

**Comment by matthiasblaesing** [comment]
I'll update VerifyLibsAndLicenses to correctly handle GPL for external binaries, that are compile-time only - give me a few days to do that.

For the performance.php: It would be great if you could see if using a different file is possible. I had a quick(!)  look at the SMF sources and there is `/Sources/Subs.php` in [SMF 2.0.15](https://download.simplemachines.org/index.php?thanks;filename=smf_2-0-15_install.zip). That are > 4000 lines of PHP code. That file would be IMHO better, because it is pure BSD:

@license http://www.simplemachines.org/about/smf/license.php BSD

so I'd take that file, add a comment where it came from, the license text and you'd be done. I assume here, that this measures the time parsing takes (no I did not look into the usage of the file :-))

I don't quire get what the changes to `html*.php` are about.

**Comment by junichi11** [comment]
> I'll update VerifyLibsAndLicenses to correctly handle GPL for external binaries, that are compile-time only - give me a few days to do that.

Thanks a lot!

> For the performance.php: It would be great if you could see if using a different file is possible. I had a quick(!) look at the SMF sources and there is /Sources/Subs.php in SMF 2.0.15. That are > 4000 lines of PHP code. That file would be IMHO better, because it is pure BSD:

I also looked at it, but I think its size is small against current performance.php(30972 lines, 837 KB) yet. Instead, we should use [nette.min.php](https://github.com/apache/incubator-netbeans/blob/fa0fe898d43e566e78ab60e9da9909631657a15e/php.editor/test/unit/data/testfiles/actions/testImportData/libs/nette.min.php)(New BSD: 22475 lines, 537 KB) or [Mpdf file](https://github.com/mpdf/mpdf/blob/development/src/Mpdf.php)(GPL: 1.01 MB) which Tomas suggested.

@tmysik What do you think?

If Mpdf file is better, I'll add it as an external file as well.

> I don't quire get what the changes to html*.php are about.

We can see "Akismet". Those codes may be part of an external script. So, at least we should not include that word. I found out the similar example code here: https://netbeans.org/bugzilla/show_bug.cgi?id=124273#c0.
```php
<body><?php
    if ($a == 1){
    ?>    
    <br/><?php                        
    $r =123;               
?><br/>
<?php
}
?>
</body>
```
 So it should be OK, maybe. Just used `<p></p>` instead of `<br/>`, `echo ""` instead of `$r = 123;`.

**Comment by tmysik** [comment]
I would definitely prefer some really big PHP file. The point is not to measure the time (that would be nice but difficult to "measure") but rather the memory; I mean to "ensure" that NB is still able to parse such a huge file after all the changes and fixes that are done in the PHP editor module.

File: php.editor/test/unit/src/org/netbeans/modules/php/editor/parser/ParserPerformanceTest.java [hunk 1]
```diff
@@ -47,17 +47,17 @@ protected void tearDown() throws Exception {
...
+        String testSource = TestUtilities.copyFileToString(testFile);
+        ASTPHP5Scanner scanner = new ASTPHP5Scanner(new StringReader(testSource));
+        ASTPHP5Parser parser = new ASTPHP5Parser(scanner);
+        Date start = new Date();
+        Symbol root = parser.parse();
+        Date end = new Date();
+        long time = end.getTime() - start.getTime();
+        System.out.println("Parsing of big files takes: " + time);
+        assertTrue(time < 2500);
+    }
```
Comment (junichi11) [inline php.editor/test/unit/src/org/netbeans/modules/php/editor/parser/ParserPerformanceTest.java]: @tmysik Thanks for you answer! OK, we should test both nette.min.php and Mpdf :)
I'm going to change to the following:
```java
    public void testBigFile_01() throws Exception {
        testBigFile("testfiles/parser/performance/Mpdf.php"); // 1.01MB
    }

    public void testBigFile_02() throws Exception {
        testBigFile("testfiles/actions/testImportData/libs/nette.min.php"); // 537KB
    }

    private void testBigFile(String filePath) throws Exception {
        File testFile = new File(getDataDir(), filePath);
        assertTrue(testFile.exists());
        String testSource = TestUtilities.copyFileToString(testFile);
        ASTPHP5Scanner scanner = new ASTPHP5Scanner(new StringReader(testSource));
        ASTPHP5Parser parser = new ASTPHP5Parser(scanner);
        Date start = new Date();
        Symbol root = parser.parse();
        Date end = new Date();
        long time = end.getTime() - start.getTime();
        System.out.println("Parsing of big files takes: " + time);
        assertTrue(time < 2500);
    }
```

**Comment by junichi11** [comment]
Fixed ParserPerformanceTest. Use Mpdf.php and nette.min.php for testing performance instead of performance.php

**Comment by junichi11** [comment]
We should be able to merge this if it is OK with @matthiasblaesing :)

**Comment by junichi11** [comment]
> Whether we can later host the binaries on apache infrastructure is a different question, but we should solve that, when it comes.

It seems that we have the mirror(http://netbeans.osuosl.org/binaries/) of hg.netbeans/binaries on the Oregon State University Open Source Lab (OSUOSL). (I don't know when it is synchronized.)

https://lists.apache.org/thread.html/80f6cb1b2077e9c491369fa8dff0cad93d5f6655d96fafc0b43130ae@%3Cdev.netbeans.apache.org%3E

**Comment by geertjanw** [comment]
Thanks, merging.

**Pull request merged**